dhis-2/dhis-services/dhis-service-reporting/src/main/java/org/hisp/dhis/validation/PeriodTypeExtended.java

```diff
@@ -81,6 +82,12 @@
 
     private boolean nonSlidingWindowsNeeded = false;
 
+    private Set<DimensionalItemId> leftSideItemIds = new HashSet<>();
+
+    private Set<DimensionalItemId> rightSideItemIds = new HashSet<>();
```

Comment:
can be `final`(?)

Reply:
Not only that, but nearly all of the private fields can be final. I've changed all that can be. Historically we've not used the final keyword much in dhis2, but it's good to convert the code as we go -- especially if we do so consistently within a module so the use of the keyword is clear. Thanks, done!